scripts/upgrade

```diff
@@ -45,6 +45,25 @@ ynh_add_config --template="cron" --destination="/etc/cron.d/$app"
 chown root: "/etc/cron.d/$app"
 chmod 644 "/etc/cron.d/$app"
 
+#=================================================
+# UPDATE APP SETTINGS
+#=================================================
+
+if [ "$upgrade_type" == "UPGRADE_APP" ]
+then
+	# Delete deprecated `firstname`/`lastname` 'null' settings. See upstream https://github.com/YunoHost/yunohost/pull/1516
+	ynh_app_setting_delete --app=$app --key=email_firstname
+	ynh_app_setting_delete --app=$app --key=email_lastname
+
+	# Retrieve admin user settings
+	email_fullname="$(ynh_user_get_info --username=$admin --key=fullname)"
+	email="$(ynh_user_get_info --username=$admin --key=mail)" # include as admin's email address is also subject to change
+
+	# Set
+	ynh_app_setting_set --app=$app --key=email_fullname --value="$email_fullname"
+	ynh_app_setting_set --app=$app --key=email --value=$email
+fi
```

Comment:
```suggestion
#=================================================
# ENSURE DOWNWARD COMPATIBILITY
#=================================================
ynh_script_progression --message="Ensuring downward compatibility..." --weight=1

ynh_app_setting_delete --app=$app --key=email_firstname
ynh_app_setting_delete --app=$app --key=email_lastname
	
email_fullname="$(ynh_user_get_info --username=$admin --key=fullname)"
email="$(ynh_user_get_info --username=$admin --key=mail)" 
```

Reply:
Thanks @ericgaspar, in 3e4c879655c6b1c2d93893915ef7951bfacc4f60 I've renamed the comment section, added in the progression info, and removed the conditional on `$upgrade_type`, but I really think it's a good idea keep the upgraded app's state in-line with a fresh install to help future development and simplify troubleshooting for users.

Reply:
We don't want to store those.
```
ynh_app_setting_set --app=$app --key=email_fullname --value="$email_fullname"
ynh_app_setting_set --app=$app --key=email --value=$email
```
We just call them when needed